### PR TITLE
 #374 Add filter talks by tag

### PR DESCRIPTION
**Pull request by deril**
Resolves [github issue](https://github.com/pivorakmeetup/pivorak-web-app/issues/374)

### Description

Filter talks by tag

### Pre-merge checklist
 
- [x] The PR relates to a single subject with a clear title and description
- [x] Verify that feature branch is up-to-date with `development` (if not - rebase it).

### CODE REVIEW

**Review by sveredyuk — DISMISSED** [review]
Well done )

File: app/controllers/talks_controller.rb [hunk 1]
```diff
@@ -8,12 +8,15 @@ def talk
   end
 
   def talks
-    @talks ||= search_against(Talk)
+    return @talks if defined? @talks
+
+    talks = search_against(Talk)
```
Comment (sveredyuk) [inline app/controllers/talks_controller.rb]: local var name same as method name... ( bad

File: app/controllers/talks_controller.rb [hunk 2]
```diff
@@ -8,12 +8,15 @@ def talk
...
       .published
-      .tagged_with(params[:tag])
       .includes(:event, :speaker)
       .page(params[:page])
       .sorted_by_date
+
+    @talks = params[:tag] ? talks.tagged_with(params[:tag]) : talks
```
Comment (sveredyuk) [inline app/controllers/talks_controller.rb]: maybe something like ?
`.public_send(params[:tag] ? :tagged_with, params[:tag] : to_a`

File: app/controllers/talks_controller.rb [hunk 3]
```diff
@@ -8,12 +8,15 @@ def talk
   end
 
   def talks
-    @talks ||= search_against(Talk)
+    return @talks if defined? @talks
```
Comment (sveredyuk) [inline app/controllers/talks_controller.rb]: definitely better to use @talks ||=

File: app/controllers/talks_controller.rb (hunk 2 above)
Comment (floor114) [reply app/controllers/talks_controller.rb]: What about method `untagged_talks`. And than

```ruby
@talks ||= params[:tag] ? untagged_talks.tagged_with(params[:tag]) : untagged_talks
```

**Review by sveredyuk — APPROVED** [review]

**Review by floor114 — APPROVED** [review]

**Pull request merged**